src/api/accounts.rs

```diff
@@ -0,0 +1,466 @@
+//! Partial implementation of the `Accounts` namespace.
+
+use crate::api::{Namespace, Web3};
+use crate::error::Error;
+use crate::helpers::CallFuture;
+use crate::types::{
+    Address, Bytes, Recovery, RecoveryMessage, SignedData, SignedTransaction, TransactionParameters, H256, U256,
+};
+use crate::Transport;
+use ethereum_transaction::{
+    Bytes as EthtxBytes, SignTransaction, SignedTransaction as EthtxSignedTransaction, Transaction,
+};
+use ethsign::{Error as EthsignError, SecretKey};
+use futures::future::{self, Either, FutureResult, Join3};
+use futures::{Async, Future, Poll};
+use parity_crypto::Keccak256;
+use std::borrow::Cow;
+use std::mem;
+
+/// `Accounts` namespace
+#[derive(Debug, Clone)]
+pub struct Accounts<T> {
+    transport: T,
+}
+
+impl<T: Transport> Namespace<T> for Accounts<T> {
+    fn new(transport: T) -> Self
+    where
+        Self: Sized,
+    {
+        Accounts { transport }
+    }
+
+    fn transport(&self) -> &T {
+        &self.transport
+    }
+}
+
+impl<T: Transport> Accounts<T> {
+    /// Gets the parent `web3` namespace
+    fn web3(&self) -> Web3<T> {
+        Web3::new(self.transport.clone())
+    }
+
+    /// Signs an Ethereum transaction with a given private key.
+    pub fn sign_transaction(&self, tx: TransactionParameters, key: &SecretKey) -> SignTransactionFuture<T> {
+        SignTransactionFuture::new(self, tx, key)
+    }
+
+    /// Hash a message according to EIP-191.
+    ///
+    /// The data is a UTF-8 encoded string and will enveloped as follows:
+    /// `"\x19Ethereum Signed Message:\n" + message.length + message` and hashed
+    /// using keccak256.
+    pub fn hash_message<S>(&self, message: S) -> H256
```

Comment:
We can now use `impl Trait` for this:
```suggestion
    pub fn hash_message(&self, message: impl AsRef<str>) -> H256 {
```

Most likely it applies to every generic method introduced in this PR. But it's not critical, just a hint :)